### PR TITLE
RavenDB-12015 - remove StoreNextMapResultId from the dispose method

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs [hunk 1]
```diff
@@ -147,6 +147,7 @@ static ReduceMapResultsBase()
             }
 
             WriteLastEtags(indexContext);
+            _mapReduceContext.StoreNextMapResultId();
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs:150]: @grisha-kotler @arekpalinski Do we need to call it also here?
https://github.com/ravendb/ravendb/blob/98d37f5a1f1754ee7062fc4523f4020a47f65cf5/src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs#L81-L85

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs:150]: The MapResultId shouldn't change if we filtered all results

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs:150]: It looks like it, but wanted to double confirm

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs:150]: Yes, we didn't do any work during the map phase.

**Pull request merged**